spark/src/test/scala/org/apache/spark/sql/delta/DDLTestUtils.scala

```diff
@@ -73,8 +74,54 @@ case class TestColumnSpec(
   }
 }
 
+object GeneratedAsIdentityType extends Enumeration {
+  type GeneratedAsIdentityType = Value
+  val GeneratedAlways, GeneratedByDefault = Value
+}
+
+case class IdentityColumnSpec(
+    generatedAsIdentityType: GeneratedAsIdentityType.GeneratedAsIdentityType,
+    startsWith: Option[Long] = None,
+    incrementBy: Option[Long] = None,
+    colName: String = "id",
+    dataType: DataType = LongType,
+    comment: Option[String] = None)
+  extends ColumnSpec {
+
+  override def ddl: String = {
+    throw new UnsupportedOperationException(
+      "DDL generation is not supported for identity columns yet")
+  }
+
+  override def structField(spark: SparkSession): StructField = {
+    var col = io.delta.tables.DeltaTable.columnBuilder(spark, colName)
+      .dataType(dataType)
+    col = (generatedAsIdentityType, startsWith, incrementBy) match {
+      case (GeneratedAsIdentityType.GeneratedAlways, Some(start), Some(step)) =>
+        col.generatedAlwaysAsIdentity(start, step)
+      case (GeneratedAsIdentityType.GeneratedAlways, Some(start), None) =>
+        col.generatedAlwaysAsIdentity(start = start, step = 1L)
+      case (GeneratedAsIdentityType.GeneratedAlways, None, Some(step)) =>
+        col.generatedAlwaysAsIdentity(start = 1L, step = step)
+      case (GeneratedAsIdentityType.GeneratedAlways, None, None) => col.generatedAlwaysAsIdentity()
+      case (GeneratedAsIdentityType.GeneratedByDefault, Some(start), Some(step)) =>
+        col.generatedByDefaultAsIdentity(start, step)
+      case (GeneratedAsIdentityType.GeneratedByDefault, Some(start), None) =>
+        col.generatedByDefaultAsIdentity(start = start, step = 1L)
+      case (GeneratedAsIdentityType.GeneratedByDefault, None, Some(step)) =>
+        col.generatedByDefaultAsIdentity(start = 1L, step = step)
+      case _ => col.generatedByDefaultAsIdentity()
+    }
```

Comment:
Could we perhaps use the default values to simplify this?
Like `val start = startsWith.getOrElse(default_value)`?